common/src/test/java/com/revenuecat/purchases/common/offlineentitlements/OfflineEntitlementsManagerTest.kt

```diff
@@ -34,19 +44,253 @@ class OfflineEntitlementsManagerTest {
+        val customerInfo = mockk<CustomerInfo>()
+        mockCalculateOfflineEntitlements(successCustomerInfo = customerInfo)
+        var customerInfoFromCallback: CustomerInfo? = null
+        offlineEntitlementsManager.calculateAndCacheOfflineCustomerInfo(
+            appUserID,
+            onSuccess = { customerInfoFromCallback = it },
+            onError = { fail("Should succeed") }
+        )
+        assertThat(customerInfoFromCallback).isEqualTo(customerInfo)
+    }
+
+    @Test
+    fun `calculateAndCacheOfflineCustomerInfo caches customer info on success`() {
+        val customerInfo = mockk<CustomerInfo>()
+        mockCalculateOfflineEntitlements(successCustomerInfo = customerInfo)
+        offlineEntitlementsManager.calculateAndCacheOfflineCustomerInfo(
+            appUserID,
+            onSuccess = {},
+            onError = { fail("Should succeed") }
+        )
+        assertThat(offlineEntitlementsManager.offlineCustomerInfo).isEqualTo(customerInfo)
+    }
+
+    @Test
+    fun `calculateAndCacheOfflineCustomerInfo only computes offline customer info one at a time and calls all success callbacks`() {
+        val customerInfo = mockk<CustomerInfo>()
+        val successCallbackSlot = slot<(CustomerInfo) -> Unit>()
+        every {
+            offlineEntitlementsCalculator.computeOfflineCustomerInfo(appUserID, capture(successCallbackSlot), any())
+        } just Runs
+        var callback1CustomerInfo: CustomerInfo? = null
+        offlineEntitlementsManager.calculateAndCacheOfflineCustomerInfo(
+            appUserID,
+            onSuccess = { callback1CustomerInfo = it },
+            onError = { fail("Should succeed") }
+        )
+        var callback2CustomerInfo: CustomerInfo? = null
+        offlineEntitlementsManager.calculateAndCacheOfflineCustomerInfo(
+            appUserID,
+            onSuccess = { callback2CustomerInfo = it },
+            onError = { fail("Should succeed") }
+        )
+        assertThat(callback1CustomerInfo).isNull()
+        assertThat(callback2CustomerInfo).isNull()
+        verify(exactly = 1) { offlineEntitlementsCalculator.computeOfflineCustomerInfo(any(), any(), any()) }
+        successCallbackSlot.invoke(customerInfo)
+        assertThat(callback1CustomerInfo).isEqualTo(customerInfo)
+        assertThat(callback2CustomerInfo).isEqualTo(customerInfo)
+        verify(exactly = 1) { offlineEntitlementsCalculator.computeOfflineCustomerInfo(any(), any(), any()) }
+    }
+
+    @Test
+    fun `calculateAndCacheOfflineCustomerInfo does not cache customer info on error`() {
+        mockCalculateOfflineEntitlements(error = PurchasesError(PurchasesErrorCode.UnknownError))
+        offlineEntitlementsManager.calculateAndCacheOfflineCustomerInfo(
+            appUserID,
+            onSuccess = { fail("Should error") },
+            onError = {}
+        )
+        assertThat(offlineEntitlementsManager.offlineCustomerInfo).isNull()
```

Comment:
I check that in the following test `calculateAndCacheOfflineCustomerInfo returns error`. I tried to minimize the things we check for on each test, but lmk if you think it would be better to check the completions on every test... It might be safer, though more verbose as well...

Reply:
I see. I always worry someone removes that test for some reason, this functionality breaks and this starts giving false positives. But I guess it's just different styles of testing so feel free to ignore

Reply:
dont worry, this is fine